queasars/job_shop_scheduling/serialization.py

```diff
@@ -0,0 +1,183 @@
+# Quantum Evolving Ansatz Variational Solver (QUEASARS)
+# Copyright 2024 DLR - Deutsches Zentrum für Luft- und Raumfahrt e.V.
+
+from json import JSONEncoder, JSONDecoder
+from typing import Any
+
+from queasars.job_shop_scheduling.problem_instances import (
+    Machine,
+    Operation,
+    Job,
+    JobShopSchedulingProblemInstance,
+    ScheduledOperation,
+    JobShopSchedulingResult,
+)
+
+
+class JSSPJSONEncoder(JSONEncoder):
+    """
+    JSONEncoder class for encoding objects of the job shop scheduling data classes in
+    queasars.job_shop_scheduling.problem_instances
+    This class can serialize:
+        Machine,
+        Operation,
+        Job,
+        JobShopSchedulingProblemInstance,
+        ScheduledOperation,
+        JobShopSchedulingResult,
+    """
+
+    def default(self, o: Any) -> Any:
+
+        if isinstance(o, tuple):
+            return {"tuple": [self.default(entry) for entry in o]}
+
+        if isinstance(o, list):
+            return {"list": [self.default(entry) for entry in o]}
+
+        if isinstance(o, dict):
+            return {"dict": self.default(list(o.items()))}
```

Comment:
Doesn't this mean that dictionaries are serialized as
```
{
  "dict": {
    "list": [
      ... entries ...
    ]
  }
}
```
That's somewhat unnecessary, no? Since this just for interal data (as JSSP will probably never be solved as a service) it doesn't matter much, though.

Reply:
You are correct. Since lists are deserialized as arrays in json anyways, i removed the "list" annotation.

Reply:
Ok thanks!